### ISSUE TITLE
Issue installing haskell_cudd (Ambiguous occurrence `unsafeForeignPtrToPtr')

### ISSUE THREAD

**Issue by maweki** [adamwalker/haskell_cudd#1]
I am running Fedora 21 64bit

I installed cudd, cudd-extras and cudd-devel via yum.
When I try `cabal install cudd` I get "Missing C library: st"
Continuing with `cabal install cudd --extra-lib-dirs=/lib64/cudd`, where Fedora installs the cudd libraries, I get 'dddmp.h not found.
Continuing with `cabal install cudd --extra-lib-dirs=/lib64/cudd --extra-include-dirs=/usr/include/cudd/`, where the header-files are installed

```
[ 1 of 11] Compiling Cudd.ForeignHelpers ( Cudd/ForeignHelpers.hs, dist/build/Cudd/ForeignHelpers.o )
Cudd/ForeignHelpers.hs:20:18:
Ambiguous occurrence `unsafeForeignPtrToPtr'
It could refer to either `Foreign.unsafeForeignPtrToPtr',
                         imported from `Foreign' at Cudd/ForeignHelpers.hs:9:1-14
                         (and originally defined in `Foreign.ForeignPtr')
                      or `Foreign.ForeignPtr.Unsafe.unsafeForeignPtrToPtr',
                         imported from `Foreign.ForeignPtr.Unsafe' at Cudd/ForeignHelpers.hs:11:1-32
                         (and originally defined in `GHC.ForeignPtr')
Failed to install cudd-0.1.0.1
```

I tried to look into that issue myself but strange enough, when I try  `cabal install --extra-lib-dirs=/lib64/cudd --extra-include-dirs=/usr/include/cudd/` within the cloned git repository, I get an `Could not find module 'Cudd.File'`-error.

Do you have any Idea how I might get the cudd-bindings running?

**Comment by adamwalker** [comment]
It is strange that you have to manually add standard lib and include directories manually. Not sure how to fix that. 

Are you using ghc 7.6? base version 4.6 exports unsafeForeignPtrToPTr from Foreign whereas later versions dont. This is probably whats causing the problem. I should have set a lower bound on the version of base to prevent this.

In that case, the solution would be to upgrade ghc. If thats not possible, I can look at making it work for older versions.

**Comment by maweki** [comment]
Yeah, that thing with the libraries is strange enough but including the paths is no real issue, I think. I was just stating those things for completeness.

I am using ghc 7.6.3 from the Fedora21 repository. Again, I wouldn't mind giving fixing it myself a shot but executing `cabal install --extra-lib-dirs=/lib64/cudd --extra-include-dirs=/usr/include/cudd/` in the haskell_cudd repository clone doesn't work with a module-not-found error for Cudd.File.

Edit:
I manually executed c2hs on File.chs which didn't work because of include errors. Then I copied all the header files from `/usr/include/cudd` into the Cudd-directory. Running c2hs compiled a hs-file so that `cabal install --extra-lib-dirs=/lib64/cudd --extra-include-dirs=/usr/include/cudd/` within the repository has the same output as the installation from hackage, `Ambiguous occurrence 'unsafeForeignPtrToPtr'`

**Comment by adamwalker** [comment]
I cloned the repo and tried the same thing. It also failed for me in the same way. I noticed that in the dist directory it hadnt built File.hs from File.chs for some reason. File.chs is definitely there so I have no idea why it cant find it.

Then, I deleted the dist directory that cabal makes at the root of the repository and tried again. This time everything worked. 

In my experience, cabal is a very flakey piece of software. Wiping the dist directory and starting again might work for you too.
